### ISSUE TITLE
Feature request: Cotanget space

### ISSUE THREAD

**Issue by GiulioRomualdi** [artivis/manif#263]
Hi @artivis, we are currently your library in our lab. We see that for each Liegroup, `manif` exposes the associated tangent space but not the cotangent one. 
I was wondering if there is a reason why it is not implemented and if there is any possible plan to add it. 

cc @riccardogrieco

**Comment by artivis** [comment]
Hi @GiulioRomualdi,
There is no plan to add this at the moment but if you'd like to discuss one we can do so. I guess the first step would be to put together some math doc to clarify the topic and set some solid foundations.
cc @joansola

**Comment by joansola** [comment]
Agree. I never digged into the cotangent space, and I am just reading about it rapidly. I have a few questions:

Given a group, is the cotangent space unique? As it is the tangent space? Or are there many possible cotangent spaces?

If it is unique, I guess we could add all its functionality, as we did with the Tangent. What would this functionality consist of? Any idea of what functions / operators / objects are needed?

And a side question: what do you use the cotangent space for?

Joan

**Comment by rebcabin** [comment]
The cotangent space seems essential for control applications, as directional derivatives (cotangent covectors) applied to the potential energy effect d'Alembert's Principle. Please see the right-most terms in Equations 16b and 16c of this paper: https://arxiv.org/abs/1609.02898.
